social_core/utils.py

```diff
@@ -108,28 +108,25 @@ def sanitize_redirect(hosts, redirect_to):
 
 def user_is_authenticated(user):
     if user and hasattr(user, 'is_authenticated'):
-        if callable(user.is_authenticated):
-            authenticated = user.is_authenticated()
-        else:
-            authenticated = user.is_authenticated
+        return (
+            user.is_authenticated()
+            if callable(user.is_authenticated)
+            else user.is_authenticated
+        )
+
     elif user:
-        authenticated = True
+        return True
     else:
-        authenticated = False
-    return authenticated
+        return False
 
 
 def user_is_active(user):
     if user and hasattr(user, 'is_active'):
-        if callable(user.is_active):
-            is_active = user.is_active()
-        else:
-            is_active = user.is_active
+        return user.is_active() if callable(user.is_active) else user.is_active
     elif user:
-        is_active = True
+        return True
     else:
```

Comment:
Else not needed here.

Reply:
I think its better if we let it, in the test case we gonna back to it

Reply:
I mean that the "else" statement is not needed, not the code.

GitHub doesn't allow me to make the proper suggestion over removed lines, but basically this (and removing the next `return False` statement):

```suggestion
    else:
    return False
```

Reply:
Fixed @nijel you could run the tests